lib/buffer.js

```diff
@@ -145,6 +145,11 @@ function assertSize(size) {
     Error.captureStackTrace(err, assertSize);
     throw err;
   }
+  if (size < 0) {
+    const err = new RangeError('"size" argument must not be negative');
+    Error.captureStackTrace(err, assertSize);
+    throw err;
+  }
```

Comment:
How about this abstraction:

```js
function assertSize(size) {
  let err = null;
  if (typeof size !== 'number') err = new TypeError('"size" argument must be a number');
  if (size < 0) err = new RangeError('"size" argument must not be negative');
  if (err) {
    // The following hides the 'assertSize' method from the
    // callstack. This is done simply to hide the internal
    // details of the implementation from bleeding out to users.
    Error.captureStackTrace(err, assertSize);
    throw err;
  }
}
```

This could improve the comments for `Error.captureStackTrace(err, assertSize);`. :-)